### PR TITLE
[HOLD][ZEPPELIN-2248] Fix javax.ws.rs and cxf version mismatch

### PR DESCRIPTION
**Pull request by jongyoul**
### What is this PR for?
Fixing version mismatch among jetty, cxf, jersey

### What type of PR is it?
[Bug Fix]

### Todos
* [x] - Match all versions of javax.ws related libraries

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-2248

### How should this be tested?
No error while starting and accessing the web at the first time

### Screenshots (if appropriate)
N/A

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: zeppelin-server/pom.xml [hunk 1]
```diff
@@ -292,10 +280,15 @@
       <version>${quartz.scheduler.version}</version>
     </dependency>
 
-    <dependency>
+    <!--<dependency>
       <groupId>com.sun.jersey</groupId>
       <artifactId>jersey-servlet</artifactId>
       <version>${jersey.servlet.version}</version>
+    </dependency>-->
```
Comment (boky01) [inline zeppelin-server/pom.xml]: Could you delete the unnecessary dependency instead of commenting out?

Comment (jongyoul) [inline zeppelin-server/pom.xml]: Thanks. I'll remove it.

**Comment by jongyoul** [comment]
I got one error from travis. It seems to be related issue

**Comment by echarles** [comment]
How does this PR relates to #932 ?
#932 simply replaces cxf with jersey2. Here I see we still have some cxf deps.

**Comment by jongyoul** [comment]
This PR doesn't remove cxf, but just fixs the version mismatch issue. I'm not sure why we have to use cxf but I also don't know what the benefit is to remove cxf. This is WIP and I will focus only the minimum fix not to raise error when starting ZeppelinServer.

**Comment by echarles** [comment]
This issue is recurring since a long time and depends IMHO on each environment (maven version...) - The main parameter to fix this is the `javax.ws.rsapi.version`.

This PR upgrades cxf from v2 to v3 and jersey from v1 to v2 which confuses me. I would rather see using only jersey2 as in #932.

Also Moving from jersey-v1 to jersey-v2 needs an update of the licence file as explained in #932.

Maybe the minimal fix would be leaving cxf as such and simply enforcing `javax.ws.rsapi.version` (I did that some time ago as temp fix).

**Comment by jongyoul** [comment]
Hold until concluding #932

**Comment by sohaibiftikhar** [comment]
Hi, what is the status on this issue. Is there some temporary fix that can be used. Please tell me if you need help fixing/testing this issue. This is important because I cannot get LDAP to work if this keeps triggering on authorisation failure.

**Comment by jongyoul** [comment]
@sohaibiftikhar Hi, I'm holding it because of the former working #932 which looks similar to this. Can you share more about your problem? I'm actually blocked by disabling shiro. Without shiro, I don't know how to handle it. Can you test this PR? I decide to hold because it's one of minor issue but you suffer from the critical problem by this issue, I can handle it sooner with your help. :-)

**Comment by sohaibiftikhar** [comment]
For some reason the jar conflict(with the no such method exception) keeps showing up for me when using shiro with `org.apache.zeppelin.realm.LdapRealm`. When I had earlier used `GroupLdapRealm` there was no issue. I am not really sure if this is the reason or it is plain chance. I can certainly help with testing it but do you know why the Jenkins test are failing with your merge request?

**Comment by jongyoul** [comment]
@sohaibiftikhar #932 didn't adopt CI then you'd better test this build and fix some minor issues from it. Thanks!

**Comment by jongyoul** [comment]
@sohaibiftikhar Sorry for confusing you. I didn't know #932 set the CI. Please test it with that PR

**Comment by jongyoul** [comment]
Will close it as resolved by #932

**Pull request closed without merging**